src/utils/number.ts

```diff
@@ -0,0 +1,7 @@
+export class NumberUtils {
+  formatViews(views: number) {
+    if (views < 1e3) return views;
+    if (views < 1e6) return `${(views / 1e3).toFixed(1)}k`;
+    return `${(views / 1e6).toFixed(1)}m`;
+  }
+}
```

Comment:
```suggestion
  formatViews(views: number) {
    if (views < 1e3) return views;
    if (views < 1e6) return `${(views / 1e3).toFixed(1)}k`;
    return `${(views / 1e6).toFixed(1)}m`;
  }
```

Này thay vì class mình nghĩ có thể dùng object `{}` thông thường hoặc export đơn thuần thôi là được.

Reply:
Cố tình dùng bạn ạ. Thống nhất vậy rùi.

Reply:
Thực ra việc xài class cũng có lí do của nó. Đầu tiên là việc re use method trực tiếp trong method thuộc class, thứ hai là việc oop hoá để gọi theo kiểu module cho dễ nhớ (chấm cái gọi ra, chứ k cần phải nhớ tên method)

Reply:
> re use method trực tiếp trong method thuộc class

Object `{}` bình thường vẫn dùng `this.xxx` được mà bác.

> việc module hoá để gọi cho dễ nhớ

`export const NumberUtils = { ... }` =))

Reply:
> Object {} bình thường vẫn dùng this.xxx được mà bác.

Hình như k đc bác, `this` ngầm hiểu phạm vi là bên trong hàm
<img width="497" alt="image" src="https://github.com/user-attachments/assets/eb8c7818-dd7e-4c70-b5ab-fed303d2dd5a">

Reply:
Để mình check lại `tsconfig` xem sao. Như code dưới đây thì bên mình typecheck không bị lỗi.

```ts
export const Utils = {
    func1() {
        console.log("func1")
    },
    func2() {
        this.func1()
    }
} 
```

Reply:
Có gì khác k bác. K thì thôi dùng class khỏi mất công commit mới.

Reply:
Hình như là lỗi [này](https://stackoverflow.com/questions/65613271/why-my-arrow-function-inside-a-object-doesnt-works) á bác. JS là vua của mọi loại ngôn ngữ lập trình :v

Mình nghĩ sẽ chọn cách class để ép mấy người contribute sau xài function thay vì arrow function, k th lại xảy ra edge case như lỗi mình nêu trên, có gì b xem giải quyết bằng tsconfig đc thì báo lại r discuss tiếp hehe

Reply:
> Có gì khác k bác. K thì thôi dùng class khỏi mất công commit mới.

Do là class này không own data, không inheritance gì cả nên mình nghĩ để raw object ok hơn, bớt overhead của class lại nữa.

> chọn cách class để ép mấy người contribute sau xài function thay vì arrow function, k th lại xảy ra edge case như lỗi mình nêu trên

Mình nghĩ này không cần thiết phải ép thế lắm, vì giờ hầu hết đầu dùng Ide/text editor có intellisense hết rồi, có lỗi thì báo hết. Nếu muốn chắc hơn nữa thì có thể setup CI/CD để check mỗi khi có PR mới luôn.

Reply:
Chủ yếu như cách mình bị á mà, k search ra này chắc cũng chả biết lỗi j:))